src/main/java/com/limechain/runtime/WasmSectionUtils.java

```diff
@@ -0,0 +1,124 @@
+package com.limechain.runtime;
+
+import com.limechain.exception.WasmRuntimeException;
+import com.limechain.runtime.version.scale.RuntimeVersionReader;
+import com.limechain.runtime.version.ApiVersions;
+import com.limechain.runtime.version.RuntimeVersion;
+import com.limechain.utils.ByteArrayUtils;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import lombok.experimental.UtilityClass;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+// TODO: [Testability concern] Maybe make this class non-static?
```

Comment:
But it already is non static, do you mean removing the `@UtilityClass` annotation?

Reply:
Yes, exactly. The two opposing points are:
1) pro: static methods are more cumbersome to mock; mocking an instance is easier (in tests, which don't exist yet)
2) con: logically, it's a static utility class (by its purpose), so instantiating it everywhere might be considered "annoying boilerplate"

Reply:
I prefer leaving it in as changing code for ease of testing isn't always a good practice.